orchestrator/careplanservice/util.go

```diff
@@ -77,3 +112,36 @@ func matchResourceIDs(request *FHIRHandlerRequest, idFromResource *string) error
 	}
 	return nil
 }
+
+// filterMatchingResourcesInBundle will find all resources in the bundle of the given type with a matching ID and return a new bundle with only those resources
+func filterMatchingResourcesInBundle(bundle *fhir.Bundle, resourceTypes []string, references []string) fhir.Bundle {
+	newBundle := fhir.Bundle{
+		Entry: []fhir.BundleEntry{},
+	}
+
+	for i, entry := range bundle.Entry {
+		var resourceInBundle coolfhir.Resource
+		err := json.Unmarshal(entry.Resource, &resourceInBundle)
+		if err != nil {
+			// We don't want to fail the whole operation if one resource fails to unmarshal
+			log.Error().Msgf("filterMatchingResourcesInBundle: Failed to unmarshal resource: %v", err)
```

Comment:
I think you could even add an OperationOutcome entry to the bundle, alternatively